### ISSUE TITLE
CMake Error Could NOT find Python (missing: Development.Module)

### ISSUE THREAD

**Issue by maximvochten** [meco-group/fatrop#20]
Issue is explained here: https://github.com/meco-group/fatrop/issues/17#issuecomment-2345839818

Temporary solution is to change [these lines in CMakeList.txt](https://github.com/meco-group/fatrop/blob/13d5860ce9866cda000117732b820b8d2b0e65ad/CMakeLists.txt#L124) back to

```
if(WITH_PYTHON)
  add_subdirectory(external/pybind11)
  add_subdirectory(fatropy)
endif()
```

**Comment by nim65s** [comment]
I can't reproduce your issue, with the following workflow:

```Dockerfile
FROM ubuntu:24.04

RUN apt-get update && apt-get install -qqy \
    build-essential \
    git \
    cmake \
    python-is-python3 \
    python3-dev

RUN git clone --recursive https://github.com/meco-group/fatrop

RUN cmake -B build -S fatrop
```

Can you provide more informations on your system, the commit you are on, and the command you use to configure the project ?

**Comment by maximvochten** [comment]
Thank you for your reply. It made me realize that the problem is because I am on Ubuntu 20.04 with CMake 3.16 (which is the latest release on Debian for Cmake). 

I discovered that "Development.Module" while finding Python was only introduced in CMake 3.18. Changing that line in CMakeLists.txt to

`find_package(Python REQUIRED COMPONENTS Interpreter Development)`

solves my issue. I also had to install pybind11 system-wide using 

`sudo apt-get install pybind11-dev`

**Comment by nim65s** [comment]
You are right, thanks. We could, either:
1. replace `Development.Module` by `Development`, as you propose, but this impose the additional requirement of having `libpython.so` available even if we don't really need it. This would prevent fatropy to be easily used with alternative python implementations like [pypy](https://pypy.org/), and also prevent fatropy to be easily distribuable on [PyPI](https://pypi.org/).
2. update the `cmake_minimum_required(VERSION 3.14)` to 3.18, but this would force ubuntu 20.04 users to upgrade either their whole system, or install an updated version of CMake (`pip install cmake` would work, but that is still an additional packaging constraint)
3. use a condition, like:
```cmake
if(CMAKE_VERSION VERSION_LESS "3.18")
  find_package(Python REQUIRED COMPONENTS Interpreter Development)
else()
  find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
endif()
```

The system-wide pybind11 is a good solution, but not the only one: you could also have used the git submodule, if did not want (or could not) run that `sudo apt …`. This is what I did with the `--recursive` flag in the example above.

**Comment by lvanroye** [comment]
Hi Guilhem and Maxim, 

Thanks a lot for figuring this out and the suggestions for fixing the issues. We  think the third option is the best solution. I implemented it (see commit bcf259d).  

Best, 
Lander
